src/validation_workflow/validation_args.py

```diff
@@ -123,22 +123,32 @@ def __init__(self) -> None:
             "--validate-digest-only",
             action="store_true",
             default=False,
-            help="(optional) Validate digest only; will not run docker to test API")
+            help="(optional) Validate digest only; will not run docker to test API",
+            dest="validate_digest_only"
+        )
         group.add_argument(
             "--using-staging-artifact-only",
             action="store_true",
             default=False,
-            help="(optional) Use only staging artifact to run docker and API test, will not validate digest")
+            help="(optional) Use only staging artifact to run docker and API test, will not validate digest",
+            dest="using_staging_artifact_only"
+        )
 
         args = parser.parse_args()
 
         if (not (args.version or args.file_path)):
             raise Exception("Provide either version number or File Path")
-        if(args.file_path):
+        if (args.file_path):
+            if 'opensearch' not in args.file_path.keys() or not set(args.file_path.keys()) <= {'opensearch', 'opensearch-dashboards'}:
+                raise Exception("Missing OpenSearch artifact details! Please provide the valid product names among opensearch and opensearch-dashboards")
             args.distribution = self.get_distribution_type(args.file_path)
             args.projects = args.file_path.keys()
-        if (('opensearch' not in args.projects)):
-            raise Exception("Missing OpenSearch OpenSearch artifact details! Please provide the same along with OpenSearch-Dashboards to validate")
+
+        if args.distribution == "docker":
+            if args.osd_build_number != "latest" and "opensearch-dashboards" not in args.projects:
+                raise Exception("Provide opensearch-dashboards in projects argument to validate OpenSearch-Dashboards")
+            if not(args.validate_digest_only or args.using_staging_artifact_only):
```

Comment:
Nit: `if not(A) and not(B)` or a better condition for understanding.
We will raise exception if both of these params are empty.

Reply:
Resolved now.